src/relations/pgbouncer_provider.py

```diff
@@ -161,6 +181,9 @@ def _on_relation_departed(self, event: RelationDepartedEvent) -> None:
         if event.departing_unit == self.charm.unit:
             self.charm.peers.unit_databag.update({self._depart_flag(event.relation): "true"})
 
+        # Check if we need to close the node port
+        self.charm.patch_port(self.external_connectivity(event))
+
```

Comment:
Do we need this in both departed and broken hooks?

Reply:
@dragomirp maybe not `departed`, but on broken hook, for sure. This allows us to close the node port if no relation is requesting to be exposed any longer.